### PR TITLE
[IOTDB-2754] SessionPool auto redirect IoTDB instance

### PR DESCRIPTION
**Pull request by CRZbulabula**

### CODE REVIEW

File: example/session/src/main/java/org/apache/iotdb/SessionPoolExample.java [hunk 1]
```diff
@@ -32,31 +32,54 @@
...
 
-  private static SessionPool pool;
+  private static SessionPool customSessionPool;
+  private static SessionPool redirectSessionPool;
   private static ExecutorService service;
 
   public static void main(String[] args)
       throws StatementExecutionException, IoTDBConnectionException, InterruptedException {
-    pool =
+    // Custom SessionPool
```
Comment (SilverNarcissus) [inline example/session/src/main/java/org/apache/iotdb/SessionPoolExample.java:41]: We should avoid commented code in example

**Review by SilverNarcissus — COMMENTED** [review]
Nice work!  We also need some test for this new session pool, could you please add some IT?

File: example/session/src/main/java/org/apache/iotdb/SessionPoolExample.java (hunk 1 above)
Comment (OneSizeFitsQuorum) [inline example/session/src/main/java/org/apache/iotdb/SessionPoolExample.java:41]: You'd better clean up the example code. Don't comment out the code...

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM. Only minor issue. PTAL

File: session/src/main/java/org/apache/iotdb/session/pool/SessionPool.java [hunk 2]
```diff
@@ -108,6 +111,20 @@ public SessionPool(String host, int port, String user, String password, int maxS
         Config.DEFAULT_CONNECTION_TIMEOUT_MS);
   }
 
+  public SessionPool(List<String> nodeUrls, String user, String password, int maxSize) {
```
Comment (OneSizeFitsQuorum) [inline session/src/main/java/org/apache/iotdb/session/pool/SessionPool.java:114]: Please update the sessionPool documentation to show greater fault tolerance when passing multiple endpoints like a session

**Review by OneSizeFitsQuorum — COMMENTED** [review]
PTAL

**Comment by CRZbulabula** [comment]
> Nice work! We also need some test for this new session pool, could you please add some IT?

Sorry, I can't add IT for this new feature. Because our IT environment only supports to run a single IoTDB Instance. In fact, this example code acts as a test. You can start IoTDB-server either on port 6667 or on port 6668 and this example code still work.

File: session/src/main/java/org/apache/iotdb/session/pool/SessionPool.java (hunk 2 above)
Comment (CRZbulabula) [reply session/src/main/java/org/apache/iotdb/session/pool/SessionPool.java:114]: Done

**Review by SilverNarcissus — APPROVED** [review]
Look good to me~

**Pull request merged**